### PR TITLE
RavenDB-13599 - Index fails to compile when using Aggregate, Join, or ThenBy

### PR DESCRIPTION
**Pull request by grisha-kotler**
https://issues.hibernatingrhinos.com/issue/RavenDB-13599

### CODE REVIEW

File: src/Raven.Server/Documents/Indexes/Static/DynamicArray.cs [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 namespace Raven.Server.Documents.Indexes.Static
 {
-    public class DynamicArray : DynamicObject, IEnumerable<object>
+    public class DynamicArray : DynamicObject, IOrderedEnumerable<object>
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Indexes/Static/DynamicArray.cs:11]: Do we need this?

File: src/Raven.Server/Documents/Indexes/Static/DynamicArray.cs [hunk 2]
```diff
@@ -335,6 +335,33 @@ public IEnumerable<dynamic> OrderByDescending(Func<IGrouping<dynamic, dynamic>,
             return new DynamicArray(_inner.Cast<DynamicGrouping>().OrderByDescending(comparable));
         }
 
+        public IOrderedEnumerable<object> CreateOrderedEnumerable<TKey>(Func<object, TKey> keySelector, IComparer<TKey> comparer, bool descending)
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Indexes/Static/DynamicArray.cs:338]: This is not used?

File: src/Raven.Server/Documents/Indexes/Static/DynamicArray.cs (hunk 1 above)
Comment (grisha-kotler) [reply src/Raven.Server/Documents/Indexes/Static/DynamicArray.cs:11]: Needed for ThenBy

Comment (grisha-kotler) [reply src/Raven.Server/Documents/Indexes/Static/DynamicArray.cs:11]: Can't do `Enumerable.ThenBy(this, ...)` without it.
It expects `IOrderedEnumerable`.

File: src/Raven.Server/Documents/Indexes/Static/DynamicArray.cs (hunk 2 above)
Comment (grisha-kotler) [reply src/Raven.Server/Documents/Indexes/Static/DynamicArray.cs:338]: Implemented because of the inheritance from `IOrderedEnumerable`

**Review by ppekrol — APPROVED** [review]

**Pull request merged**